src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -0,0 +1,277 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Arcus.Messaging.Abstractions;
+using Arcus.Messaging.Pumps.Abstractions;
+using GuardNet;
+using Microsoft.Azure.ServiceBus;
+using Microsoft.Azure.ServiceBus.Core;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Arcus.Messaging.Pumps.ServiceBus
+{
+    public abstract class AzureServiceBusMessagePump<TMessage> : MessagePump<TMessage, AzureServiceBusMessageContext>
```

Comment:
It's propably me, but could you explain some exception handling strategy that is used here? What if child throws, parent throws, invalid deserailziation, no transaction ID found in message, ...

Reply:
It's being logged here, but open for consumers to extend how they want to.
https://github.com/arcus-azure/arcus.messaging/pull/32/files/5688f51056f7a1d7348d9187b36ded62c3d6606c#diff-8676b3d1a6cbfabcf514f0b762f99764R54

Over time it should get enriched with contextual information.

Reply:
Is that ok for you or do I need to take action?

Reply:
Ah, yes OK; The exception handling method is used in the messageReceiver options; but maybe we can use it ourselves also?
Like, try/catch in the `MessagePump` for the deserialization and also pass allong the exception to that method? Just thinking out loud, because having this virtual method in the `MessagePump` made me think that the functionality of the messagePump _itself_ had also an exception handling strategy.
Follow me?

Reply:
Not really but I've improved it a bit in https://github.com/arcus-azure/arcus.messaging/pull/32/commits/f8beb4618a9600d2b9b2f1743af885dba63a3c5a, think this is a bit more what you're looking for right?

Reply:
Yep, bit like that. Thanks.